packages/backend/migration/1644010796173-convert-hard-mutes.js

```diff
@@ -0,0 +1,49 @@
+const RE2 = require('re2');
+const { MigrationInterface, QueryRunner } = require("typeorm");
+
+module.exports = class convertHardMutes1644010796173 {
+    name = 'convertHardMutes1644010796173'
+
+    async up(queryRunner) {
+        (await queryRunner.query(`SELECT "userId", "mutedWords" FROM "user_profile"`)).forEach(entry => {
+                let words = entry.mutedWords
+                        .map(line => {
+                                const regexp = line.join(" ").match(/^\/(.+)\/(.*)$/);
+                                if (regexp) {
+                                        // convert regexp's
+                                        try {
+                                                new RE2(regexp[1], regexp[2]);
+                                                return `/${regexp[1]}/${regexp[2]}`;
+                                        } catch (err) {
+                                                // invalid regex, ignore it
+                                                return [];
+                                        }
+                                } else {
+                                        // remove empty segments
+                                        return line.filter(x => x !== '');
+                                }
+                        })
+                        // remove empty lines
+                        .filter(x => !(Array.isArray(x) && x.length === 0));
+
+                queryRunner.query(`UPDATE "user_profile" SET "mutedWords" = '${JSON.stringify(words)}'::jsonb WHERE "userId" = '${entry.userId}'`);
```

Comment:
ここでSQLインジェクションが可能にならないか不安があります👀

Reply:
do you think this would be enough?
```suggestion
                let json = let json = JSON.stringify(words).replaceAll("'", "''");
                queryRunner.query(`UPDATE "user_profile" SET "mutedWords" = '${json}'::jsonb WHERE "userId" = '${entry.userId}'`);
```

Reply:
I gave TypeORM's docs a quick check, and from what I gather this is how it should be done:

```suggestion
                queryRunner.query(`UPDATE "user_profile" SET "mutedWords" = ':json'::jsonb WHERE "userId" = ':userId'`, { json: json, :userId = entry.userId });
```

From here: https://typeorm.io/#/select-query-builder/using-parameters-to-escape-data

Reply:
I tried that before, but that does not work because `queryRunner` is a `QueryRunner` and not a `SelectQueryBuilder`.

Reply:
I found a way to get a query builder from a query runner.